### DISCUSSION TITLE
User Defined Material for Mixed Mode Phase Field Fracture Model

### DISCUSSION THREAD

**Discussion by DanielChou0916** [idaholab/moose#27658] [Q&A Modules: Solid mechanics]
Dear MOOSE community

I'm trying to apply a more complex strain energy decomposition approach in phase field fracture model to capture the rock-like material, which is governed by additional variables such as internal friction angle and cohesion.
The explicit decompositions for the material 'ComputeLinearElasticPFFractureStress' is scripted in:
..\projects\moose\modules\solid_mechanics\src\materials\ComputeLinearElasticPFFractureStress.C

And this file includes the .h file ''ComputeLinearElasticPFFractureStress.h'' in the directory:
..\projects\moose\modules\solid_mechanics\include\materials\ComputeLinearElasticPFFractureStress.h
As well as two sub-sequence inclusions 
''ComputePFFractureStressBase.h'' and "ComputeStressBase.h" from the same folder:
..\projects\moose\modules\solid_mechanics\include\materials\ComputePFFractureStressBase.h
and
..\projects\moose\modules\solid_mechanics\include\materials\ComputeStressBase.h

I'm wondering if I want to use the strain decomposition method which is not included in "ComputeLinearElasticPFFractureStress.C".
To aviod messing up, I'm planning to create a new MyPPFractureStress.C file, and set up additional inputs and outputs for this, but I would like to know that
will I need to also create several new .h files to handle MyPPFractureStress.C?

Does anyone has similar experience and is willing to share with me?
I really appreciate for your patience and time :)

Sincerely

**Comment by GiudGiud** [comment]
@1runer
